app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/BaseAppsmithRepositoryImpl.java

```diff
@@ -139,20 +139,34 @@ protected DBObject getDbObject(Object o) {
                 });
     }
 
+    public Mono<UpdateResult> updateById(String id, Update updateObj, AclPermission permission) {
+        if (id == null) {
+            return Mono.error(new AppsmithException(AppsmithError.INVALID_PARAMETER, FieldName.ID));
+        }
+        return ReactiveSecurityContextHolder.getContext()
+                .map(ctx -> ctx.getAuthentication())
+                .map(auth -> auth.getPrincipal())
+                .flatMap(principal -> {
+                    User user = (User) principal;
+                    Query query = new Query(Criteria.where("id").is(id));
+                    query.addCriteria(new Criteria().andOperator(notDeleted(), userAcl(user, permission)));
+                    return mongoOperations.updateFirst(query, updateObj, this.genericDomain);
+                });
+    }
+
     protected Mono<T> queryOne(List<Criteria> criterias, AclPermission aclPermission) {
         return ReactiveSecurityContextHolder.getContext()
                 .map(ctx -> ctx.getAuthentication())
                 .flatMap(auth -> {
                     User user = (User) auth.getPrincipal();
                     return mongoOperations.query(this.genericDomain)
-                            .matching(createQueryWithPermission(criterias, auth,aclPermission))
+                            .matching(createQueryWithPermission(criterias, (User) auth.getPrincipal(), aclPermission))
```

Comment:
```suggestion
                            .matching(createQueryWithPermission(criterias, user, aclPermission))
```
Nit: We have already created an object for `User`. Should we be using that object itself?